oci/layout/oci_archive_dest.go

```diff
@@ -0,0 +1,126 @@
+	return ociImgDest.SupportedManifestMIMETypes()
+}
+
+// SupportsSignatures returns an error (to be displayed to the user) if the destination certainly can't store signatures
+func (d *ociArchiveImageDestination) SupportsSignatures() error {
+	return ociImgDest.SupportsSignatures()
+}
+
+// ShouldCompressLayers returns true iff it is desirable to compress layer blobs written to this destination
+func (d *ociArchiveImageDestination) ShouldCompressLayers() bool {
+	return ociImgDest.ShouldCompressLayers()
+}
+
+// AcceptsForeignLayerURLs returns false iff foreign layers in manifest should be actually
+// uploaded to the image destination, true otherwise.
+func (d *ociArchiveImageDestination) AcceptsForeignLayerURLs() bool {
+	return ociImgDest.AcceptsForeignLayerURLs()
+}
+
+// MustMatchRuntimeOS returns true iff the destination can store only images targeted for the current runtime OS. False otherwise
+func (d *ociArchiveImageDestination) MustMatchRuntimeOS() bool {
+	return ociImgDest.MustMatchRuntimeOS()
+}
+
+// PutBlob writes contents of stream and returns data representing the result (with all data filled in).
+// inputInfo.Digest can be optionally provided if known; it is not mandatory for the implementation to verify it.
+// inputInfo.Size is the expected length of stream, if known.
+func (d *ociArchiveImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobInfo) (types.BlobInfo, error) {
+	return ociImgDest.PutBlob(stream, inputInfo)
+}
+
+// HasBlob returns true iff the image destination already contains a blob with the matching digest which can be reapplied using ReapplyBlob
+func (d *ociArchiveImageDestination) HasBlob(info types.BlobInfo) (bool, int64, error) {
+	return ociImgDest.HasBlob(info)
+}
+
+func (d *ociArchiveImageDestination) ReapplyBlob(info types.BlobInfo) (types.BlobInfo, error) {
+	return ociImgDest.ReapplyBlob(info)
+}
+
+// PutManifest writes manifest to the destination.
+func (d *ociArchiveImageDestination) PutManifest(m []byte) error {
+	return ociImgDest.PutManifest(m)
+}
+
+func (d *ociArchiveImageDestination) PutSignatures(signatures [][]byte) error {
+	return ociImgDest.PutSignatures(signatures)
+}
+
+// Commit marks the process of storing the image as successful and asks for the image to be persisted
+// after the directory is made, it is tarred up into a file and the directory is deleted
+func (d *ociArchiveImageDestination) Commit() error {
+	if err := ociImgDest.Commit(); err != nil {
+		return errors.Wrapf(err, "error storing image %q", ociRef.(ociReference).tag)
+	}
+
+	// path of directory to tar up
+	path := ociRef.(ociReference).resolvedDir
+	if err := tar(path); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// tar converts the direstory at path to a tar file and deletes the directory
+func tar(path string) error {
+	// temporary path of the tar file
+	tempPath := path + "1"
+	defer os.Rename(tempPath, path)
+	defer os.RemoveAll(path)
```

Comment:
It would be somewhat nice if the code allowed replacing an older tar _file_ in the destination location atomically with a new image. It’s by no means required _now_, but it shouldn’t be impossible.

And even disregarding atomicity, it is at least somewhat plausible that the user has used exactly the same destination path in the past, and a file already exists there, so we can’t just create a directory in the same path, at least without removing the old contents. 

Overall, sharing the path for both the file and directory, though kinda a neat trick, is just not worth the hassle AFAICT.